### PR TITLE
Update kruize config to include cloudwatch service

### PR DESCRIPTION
**Pull request by kusumachalasani**
## Why do we need this change? :thought_balloon:

Include the ENV changes of Kruize to include Cloudwatch configuration.

## Documentation update? :memo:

- [ ] Yes
- [x] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [ ] Does this change depend on specific version of Kruize
  - If yes what is the version no: <Yet to release>
  - [ ] Is that image available in production or needs deployment?
- [ ] Bugfix
- [x] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

### CODE REVIEW

File: scripts/kruizecloudwatchcofig.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "cloudwatch": {
+    "accessKeyId": "",
+    "logGroup": "",
+    "region": "",
+    "secretAccessKey": ""
+  }
+}
```
[the remark this answers is not in the document]
Comment (kusumachalasani) [reply scripts/kruizecloudwatchcofig.json:8]: The configuration format employed by kruize resembles the following. Given that the ROS format incorporates a two-level structure to get the cloudwatch config, kruize cannot directly interpret this configuration.
 To align with the kruize format, I've generated a new file with the appropriate structure for kruize's interpretation. Alternatively, we might consider integrating the necessary format into the existing configuration file.
```
kruizeconfigjson: |
    {
      "clustertype":"kubernetes",
      "k8stype" : "minikube",
      "authtype": "",
      "monitoringagent":"prometheus",
      "monitoringservice":"prometheus-k8s",
      "monitoringendpoint":"prometheus-k8s",
      "savetodb":"true",
      "dbdriver":"jdbc:postgresql://",
      "hibernate": {
        "dialect": "org.hibernate.dialect.PostgreSQLDialect",
        "driver": "org.postgresql.Driver",
        "c3p0minsize": 2,
        "c3p0maxsize": 5,
        "c3p0timeout": 300,
        "c3p0maxstatements": 50,
        "hbm2ddlauto": "update",
        "showsql": "true"
      },
      "cloudwatch": {
        "accessKeyId": "",
        "secretAccessKey": "",
        "region": "",
        "logGroup": "kruize-logs",
        "logStream": "kruize-stream",
        "logLevel": "INFO"
      }
    }
    ```

[the remark this answers is not in the document]
Comment (patilsuraj767) [reply scripts/kruizecloudwatchcofig.json:8]: It is not under our control, We will not be able to provide a json file in kruize format. [cdappconfig.json ](https://github.com/RedHatInsights/ros-ocp-backend/blob/main/scripts/cdappconfig.json) file and `ACG_CONFIG` env variable is set directly by clowder operator.

[the remark this answers is not in the document]
Comment (kusumachalasani) [reply scripts/kruizecloudwatchcofig.json]: Updated the structure in kruize accordingly.

File: kruize-clowdapp.yaml [hunk 2]
```diff
@@ -264,3 +268,7 @@ parameters:
   value: "info"
 - name: KRUIZE_PARTITION_INTERVAL
   value: "0 0 * * *" # Run the cronjob every day at midnight. Earlier it was set for 16th day
+- name: KRUIZE_CW_LOG_STREAM
+  value: "kruize-stream"
```
Comment (saltgen) [inline kruize-clowdapp.yaml:272]: @kusumachalasani Please update the stream value to `kruize-recommendations` i.e. the Openshift deployment name, the current one seems a bit generic.

**Review by saltgen — CHANGES_REQUESTED** [review]

**Review by saltgen — DISMISSED** [review]
Thank you for the changes @kusumachalasani! LGTM 👍🏼

**Comment by saltgen** [comment]
@kusumachalasani Since Kruize is upstream I think the value for CW log group the same should be configurable similar to log stream name.
In case the value is hard-coded in Kruize codebase atm, do you prefer the same for now?

**Comment by kusumachalasani** [comment]
> @kusumachalasani Since Kruize is upstream I think the value for CW log group the same should be configurable similar to log stream name. In case the value is hard-coded in Kruize codebase atm, do you prefer the same for now?

All the variables (including logGroup and logStream) related to CW logging are configurable in kruize upstream.  If it is not configured through both json and ENV variables, it picks up the default ones. In kruize-clowdapp.yaml I didn't mention logGroup as part of an ENV because Suraj mentioned that it cannot be changed and will be picked up from the clowder json.

**Comment by saltgen** [comment]
> > @kusumachalasani Since Kruize is upstream I think the value for CW log group the same should be configurable similar to log stream name. In case the value is hard-coded in Kruize codebase atm, do you prefer the same for now?
> 
> All the variables (including logGroup and logStream) related to CW logging are configurable in kruize upstream. If it is not configured through both json and ENV variables, it picks up the default ones. In kruize-clowdapp.yaml I didn't mention logGroup as part of an ENV because Suraj mentioned that it cannot be changed and will be picked up from the clowder json.

I don't see an issue if the group is mentioned explicitly, although it should work fine if the defaults are picked up from something like `clowder.LoadedConfig` and then used for log streaming.
Anyways, as far as changes on ROS side are concerned, we are good.

**Review by saltgen — APPROVED** [review]

**Pull request merged**